include/numerics/sparse_matrix.h

```diff
@@ -415,6 +415,23 @@ class SparseMatrix : public ReferenceCountedObject<SparseMatrix<T>>,
                          false); // false means DO NOT REUSE submatrix
   }
 
+  /**
+   * Similar to the above function, this function creates a \p
+   * submatrix which is defined by the indices given in the \p rows
+   * and \p cols vectors.
+   * Note: Both \p rows and \p cols can be unsorted;
+           Use the above function for better efficiency if your indices are sorted;
+   *       \p rows and \p cols contain indices that are owned by all processors.
+   */
+  virtual void create_submatrix_nosort(SparseMatrix<T> & submatrix,
+                                         const std::vector<numeric_index_type> & rows,
+                                         const std::vector<numeric_index_type> & cols) const
+  {
+    this->_get_submatrix_nosort(submatrix,
```

Comment:
Hmm... I can't remember currently why we did this "public function calling a protected virtual" design previously, it may have been something to do with backwards compatibility or something... I see no reason to propagate this strange idiom unnecessarily, `create_submatrix_nosort()` can just be added as a virtual `libmesh_not_implemented()` function at the `SparseMatrix` level without this additional indirection.

Reply:
Thanks @jwpeterson . I agree and I will make the change.